Add blog post: Introducing Proxy Mode (#49)
tlsnotary/website: blog/2026-04-22-proxy-mode/2026-03-23-proxy-mode/index.md

<Figure
  src={require('./light/mpc_mode.svg').default}
  darkSrc={require('./dark/mpc_mode.svg').default}
  caption="MPC-TLS: the prover connects to the server. Prover and verifier jointly perform the TLS handshake."
  width={500}
/>

In proxy mode, the **verifier** connects to the server and forwards encrypted traffic. The server sees the verifier's IP address, not the prover's.

<Figure
  src={require('./light/proxy_mode.svg').default}
  darkSrc={require('./dark/proxy_mode.svg').default}
  caption="Proxy mode: the verifier connects to the server, forwards encrypted TLS traffic, and verifies a ZK proof after the session."
  width={500}
/>

| | MPC-TLS | Proxy Mode |
|---|---|---|
| **Trust model** | No trusted party; security from MPC | Verifier must trust its own network path to the server |
| **Who connects to the server** | The prover | The verifier (as proxy) |
| **Detectable by server** | No, looks like normal TLS | Yes, the server sees the verifier's IP |
| **Bandwidth (prover-verifier)** | High (MPC protocol overhead) | Low (only forwarded TLS packets + ZK proof) |
| **Speed** | Slower | Significantly faster |

> | **Blind verification** | Possible | No, verifier connects to the server directly |

Reviewer: I understand this more as blind notarization possible, because blind verification is impossible, no?

Author: I wanted to avoid using the word notarization here :-)

## Who Connects to the Server Matters

A key difference between the two modes is which party opens the TCP connection to the web server.

In **MPC-TLS**, the prover's device connects directly. From the server's perspective, this looks like a normal request from a regular user. The server has no way to detect that TLSNotary is involved.

In **proxy mode**, the verifier connects to the server on behalf of the prover. This has practical consequences:

- **Server-side blocking.** Many web applications block requests from known cloud or data center IPs to prevent bots and scraping. If the verifier runs on a server, its IP may be flagged. A common workaround is routing through a residential proxy, but this introduces another intermediary and further weakens the network trust assumptions.
- **No blocking issue with MPC-TLS.** Because the prover's own device connects to the server, there is no risk of IP-based blocking. The request originates from the user's residential or mobile connection, exactly as the server expects.

> This is another reason MPC-TLS is the stronger choice for high-stakes use cases: it avoids the detectability and censorship concerns entirely.

Reviewer: ```suggestion
This is another advantage of MPC-TLS: it avoids the detectability and censorship concerns entirely.
```
